### PR TITLE
[T3A5][T16-A1]Lai Zhiyuan

### PR DESCRIPTION
**Pull request by zyjames**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/ui/Formatter.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	
+	public static String getLineSeparator() {
+		return LINE_SEPARATOR;
+	}
+	
+	public static String getDivider() {
+		return DIVIDER;
+	}
+	
+	/**
```
Comment (okkhoy) [inline src/seedu/addressbook/ui/Formatter.java]: indentation is off here!

File: src/seedu/addressbook/ui/Formatter.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            displayIndex++;
+        }
+        return formatted.toString();
+    }
+    
+    /**
+     * Shows a list of persons to the user, formatted as an indexed list.
+     * Private contact details are hidden.
+     */
+    public static List<String> showPersonListView(List<? extends ReadOnlyPerson> persons) {
```
Comment (okkhoy) [inline src/seedu/addressbook/ui/Formatter.java]: perhaps you need a better name than `showPersonListView`; you are not really showing to the user, you are merely formatting and adding it to the list, no?

File: src/seedu/addressbook/ui/TextUi.java [hunk 3]
```diff
@@ -17,24 +16,9 @@
...
-    private static final String DIVIDER = "===================================================";
-
-    /** Format of indexed list item */
-    private static final String MESSAGE_INDEXED_LIST_ITEM = "\t%1$d. %2$s";
-
-
     /** Offset required to convert between 1-indexing and 0-indexing.  */
     public static final int DISPLAYED_INDEX_OFFSET = 1;
 
-    /** Format of a comment input line. Comment lines are silently consumed when reading user input. */
```
Comment (okkhoy) [inline src/seedu/addressbook/ui/TextUi.java]: possibly displayed index offset can also go to the formatter?

File: src/seedu/addressbook/ui/TextUi.java [hunk 4]
```diff
@@ -75,7 +59,7 @@ private boolean isCommentLine(String rawInputLine) {
      * @return command (full line) entered by the user
      */
     public String getUserCommand() {
-        out.print(LINE_PREFIX + "Enter command: ");
+        out.print(Formatter.getLinePrefix() + "Enter command: ");
```
Comment (okkhoy) [inline src/seedu/addressbook/ui/TextUi.java]: a better way would be to pass the string to the formatter, and accept a completely formatted string.
in the current form, you are still doing formatting and printing together, no?

**Review by okkhoy — COMMENTED** [review]
isn't it still better to pass the message/content to be displayed to the formatter, get the formatted string and simply display that in the UI class? think about it.

**Comment by okkhoy** [comment]
@zyjames 
Some comments added. Please ack & close the PR after reading comments.

File: src/seedu/addressbook/ui/Formatter.java (hunk 2 above)
Comment (zyjames) [reply src/seedu/addressbook/ui/Formatter.java:82]: Thanks for pointing out the mistakes!(:

File: src/seedu/addressbook/ui/TextUi.java (hunk 3 above)
Comment (zyjames) [reply src/seedu/addressbook/ui/TextUi.java:35]: I thought like the displayed index offset is part of UI, thus i did not separate it out.

File: src/seedu/addressbook/ui/TextUi.java (hunk 4 above)
Comment (zyjames) [reply src/seedu/addressbook/ui/TextUi.java:62]: True, will think about how to further improve this!(: Thanks!

**Pull request closed without merging**